include/mav/Message.h

```diff
@@ -222,6 +252,14 @@ namespace mav {
             return Header<uint8_t*>(_backing_memory.data());
         }
 
+        [[nodiscard]] const Signature<const uint8_t*> signature() const {
+            return Signature<const uint8_t*>(&_backing_memory[MessageDefinition::HEADER_SIZE + header().len() + MessageDefinition::CHECKSUM_SIZE]);
```

Comment:
The signature accessor will only be valid when the packet was finalized, otherwise the `header().len()` call will be incorrect. Can you add a check for this? E.g. if we don't know the offset of the CRC, we can not possibly know the location of the signature.

Reply:
Throw an exception

Reply:
Done!